### PR TITLE
Match Pythran interface when generating index operator

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Otherwise calls to ``a[i,j]`` are turned into ``a[(i,j)]`` in C++, ``,`` being the sequence operator.

### CODE REVIEW

**Comment by scoder** [comment]
Hmm, that can't really be all. There's probably a second part of this change missing in `ExprNodes.py`, around line 4197 and/or 4228, which call `pythran_indexing_code()` to generate the actual index access.

**Comment by scoder** [comment]
Yeah, I tried that, too (and that specific change is actually best applied inside of `pythran_indexing_code()`), but it gives me invalid C++ code for the `numpy_pythran` test:
```
numpy_pythran.cpp: In function ‘PyObject* __pyx_pf_13numpy_pythran_6_diffuse_numpy(PyObject*, pythonic::types::ndarray<double, 2ul>, int)’:
numpy_pythran.cpp:3992:173: error: ‘::type’ has not been declared
   typename std::remove_reference<decltype(std::declval<pythonic::types::ndarray<double,2>>()(pythonic::types::contiguous_slice(0,0),pythonic::types::contiguous_slice(0,0))>::type __pyx_t_10;
```
The previous declaration (in current Cython master) was:
```
   typename std::remove_reference<decltype(std::declval<pythonic::types::ndarray<double,2>>()[pythonic::types::contiguous_slice(0,0),pythonic::types::contiguous_slice(0,0)])>::type __pyx_t_10;
```
You can directly run the pythran test like this:
```
CFLAGS="-Wall -O0 -ggdb" python runtests.py -vv --backend=cpp --debug pythran
```
I do not know how the C++ code is supposed to look like here, so I'm not in a position to fix it.

File: Cython/Compiler/Pythran.py [hunk 1]
```diff
@@ -76,7 +76,8 @@ def index_code(idx):
         raise ValueError("unsupported indexing type %s!" % idx.type)
 
     indexing = ",".join(index_code(idx) for idx in indices)
-    return type_remove_ref("decltype(std::declval<%s>()[%s])" % (pythran_type(type_), indexing))
+    index = '[%s]' if len(indices) == 1 else '(%s)'
+    return type_remove_ref(("decltype(std::declval<%s>()" + index) % (pythran_type(type_), indexing))
```
Comment (scoder) [inline Cython/Compiler/Pythran.py]: Ah, there was a closing parenthesis missing here, after the `index`. With that, it's now compiling and giving the correct result for me.

**Comment by serge-sans-paille** [comment]
\o/ thanks!
I'll activate result checking in pythran tests suite once this commit is in a released cython.

**Pull request closed without merging**